OpenRA.Mods.Common/Traits/Buildings/PrimaryBuilding.cs

```diff
@@ -99,17 +102,25 @@ public void SetPrimaryProducer(Actor self, bool isPrimary)
 								a.Actor != self &&
 								a.Actor.Owner == self.Owner &&
 								a.Trait.IsPrimary &&
```

Comment:
Adding `a.Trait.Info.ProductionQueues.Intersect(Info.ProductionQueues).Any()` here should fix the issue stopping you from spliting PrimaryBuilding status.